tooling/minimize/src/main.rs

```diff
@@ -143,6 +146,14 @@ fn split_args(args: Args) -> (Vec<String>, Vec<String>) {
     (minimize_args, rustc_args)
 }
 
+fn get_run_program(args: Vec<String>) -> fn(miniutil::Program) -> miniutil::TerminationInfo {
```

Comment:
You really like function pointers, don't you? ;)
But I don't think making this higher-order is helpful. You can just do the case distinction later -- i.e., make this function actually run the program, according to the flags.